### PR TITLE
python311Packages.orbax-checkpoint: 0.5.5 -> 0.5.7

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

cc @fabaff 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 300913` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.distrax</li>
    <li>python311Packages.distrax.dist</li>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>python311Packages.dalle-mini</li>
    <li>python311Packages.dalle-mini.dist</li>
    <li>python311Packages.dm-haiku</li>
    <li>python311Packages.dm-haiku.dist</li>
    <li>python311Packages.flax</li>
    <li>python311Packages.flax.dist</li>
    <li>python311Packages.orbax-checkpoint</li>
    <li>python311Packages.orbax-checkpoint.dist</li>
    <li>python311Packages.vqgan-jax</li>
    <li>python311Packages.vqgan-jax.dist</li>
    <li>python312Packages.orbax-checkpoint</li>
    <li>python312Packages.orbax-checkpoint.dist</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
Change looks good to me.

File: pkgs/development/python-modules/orbax-checkpoint/default.nix [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
 , numpy
 , protobuf
-, pytest-xdist
-, pytestCheckHook
-, pythonOlder
 , pyyaml
 , tensorstore
 , typing-extensions
+, pytest-xdist
+, pytestCheckHook
```
Comment (samuela) [inline pkgs/development/python-modules/orbax-checkpoint/default.nix:20]: were any arguments added or removed here? if not, let's avoid polluting the git blame

File: pkgs/development/python-modules/orbax-checkpoint/default.nix [hunk 2]
```diff
@@ -1,43 +1,43 @@
...
+    hash = "sha256-3hRUm4mSIKT0RUU5Z8GsLXFluBUlM0JYd0YAXwOpgTs=";
   };
 
-  nativeBuildInputs = [
+  build-system = [
     flit-core
   ];
 
-  propagatedBuildInputs = [
+  dependencies = [
```
Comment (samuela) [inline pkgs/development/python-modules/orbax-checkpoint/default.nix:40]: are these new? i've never seen them before -- are there docs to checkout for them?

if they aren't necessary for the upgrade, i propose that we leave the diff to a separate PR

File: pkgs/development/python-modules/orbax-checkpoint/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply pkgs/development/python-modules/orbax-checkpoint/default.nix]: It was for re-ordering according to the coding-style.
I removed that from this PR.

File: pkgs/development/python-modules/orbax-checkpoint/default.nix (hunk 2 above)
Comment (GaetanLepage) [reply pkgs/development/python-modules/orbax-checkpoint/default.nix]: Yes, they are new (from ~2 months ago).

https://github.com/NixOS/nixpkgs/commit/4d0cca465476a1142c187198c9fe3fcd89aec6d1
https://github.com/NixOS/nixpkgs/commit/b9138b7c072f55dcfd7b8c0747c3c4291bd0b28e

However, I removed them from this PR to not clutter the git blame.

**Review by samuela — APPROVED** [review]
LGTM!
